### PR TITLE
(nixos/calibre-server): Add new http & auth options

### PR DESCRIPTION
**Pull request by gaelreyrol**
###### Description of changes

It is my first contribution to the **nixpkgs** repository and I hope my changes are not that bad!

This PR adds 4 new options to the `calibre-server` service. It will allow to configure host interface & port and auth configuration.

I also add a test for a simple custom case.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gaelreyrol** [comment]
@GrahamcOfBorg test calibre-server

**Review by drupol — APPROVED** [review]

**Review by drupol — APPROVED** [review]

File: nixos/modules/services/misc/calibre-server.nix [hunk 1]
```diff
@@ -26,44 +37,93 @@ in
...
+  config = mkIf cfg.enable {
 
+    systemd.services.calibre-server = {
+      description = "Calibre Server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        User = cfg.user;
+        Restart = "always";
+        ExecStart = "${pkgs.calibre}/bin/calibre-server ${lib.concatStringsSep " " cfg.libraries} ${execFlags}";
```
Comment (drupol) [inline nixos/modules/services/misc/calibre-server.nix:122]: I would add the `package` option use something like:

```
${lib.getExe cfg.package}
```

See example in here: https://github.com/NixOS/nixpkgs/pull/237824

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by gaelreyrol** [comment]
@GrahamcOfBorg test calibre-server

File: nixos/modules/services/misc/calibre-server.nix [hunk 2]
```diff
@@ -18,52 +29,100 @@ in
...
-          ExecStart = "${pkgs.calibre}/bin/calibre-server ${lib.concatStringsSep " " cfg.libraries}";
-        };
-
+      description = "Calibre Server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        User = cfg.user;
+        Restart = "always";
+        ExecStart = "${cfg.package}/bin/calibre-server ${lib.concatStringsSep " " cfg.libraries} ${execFlags}";
```
Comment (drupol) [inline nixos/modules/services/misc/calibre-server.nix:121]: I don't know if the `/bin/calibre-server` is required

**Review by drupol — APPROVED** [review]

File: nixos/modules/services/misc/calibre-server.nix (hunk 2 above)
Comment (gaelreyrol) [reply nixos/modules/services/misc/calibre-server.nix:121]: The package `calibre` contains multiple executables and `calibre-server` is not the default one.

Comment (drupol) [reply nixos/modules/services/misc/calibre-server.nix:121]: Fair enough. If needed, the meta attribute `mainProgram` can be set in the derivation so it's not needed here.

**Comment by gaelreyrol** [comment]
@GrahamcOfBorg test calibre-server

**Review by drupol — APPROVED** [review]

**Pull request merged**